kafkabp/config.go

```diff
@@ -0,0 +1,61 @@
+package kafkabp
+
+import (
+	"github.com/Shopify/sarama"
+)
+
+// ConsumerConfig can be used to configure a kafkabp Consumer.
+//
+// Can be deserialized from YAML.
+//
+// Example:
+//
+// kafka:
+//   brokers:
+//     - 127.0.0.1:9090
+//     - 127.0.0.2:9090
+//   topic: sample-topic
+//   clientID: myclient
+//   offset: oldest
+type ConsumerConfig struct {
+	// Required. Brokers specifies a slice of broker addresses.
+	Brokers []string `yaml:"brokers"`
+
+	// Required. Topic is used to specify the topic to consume.
+	Topic string `yaml:"topic"`
+
+	// Optional. A user-provided string sent with every request to the brokers
```

Comment:
Saying this is optional is a little bit misleading.

In ads-go we have two different consumer implementations. The `GroupConsumer` is for the more traditional, delivery every message once queue consumer (when every consumer in the same group have the same client id). This one doesn't do that. This consumer is for the use cases of using kafka to send some config/data to all the running instances use-case, so we actually want every consumer to have the whole view of the kafka topic. As a result, we always need to make sure that every consumer has an unique client id (if you give a group of pods the same client id with this consumer, they won't actually make sure that every kafka message is consumed by exactly one pod, and also every pod won't have the whole view of the kafka topic either).

I'd say this is required, and the comment should mention that every consumer should have an unique `ClientID`. for k8s pods, the k8s pod id is usually a good candidate to be used as the `ClientID`.

Reply:
If we are going to share the same `ConsumerConfig` between `NewConsumer` and `NewGroupConsumer`, then I guess this is actually optional for `NewGroupConsumer`, but still using the default value for `NewGroupConsumer` is not really recommended.